src/System.Linq.Parallel/tests/ExchangeTests.cs

```diff
@@ -131,22 +132,28 @@ public static void Merge_Ordered_Longrunning(Labeled<ParallelQuery<int>> labeled
         }
 
         [Theory]
-        [MemberData("MergeData", (object)(new int[] { 16, 1024 }))]
-        [MemberData("ThrowOnCount_AllMergeOptions_MemberData", (object)(new int[] { 16, 1024 }))]
+        [MemberData("ThrowOnCount_AllMergeOptions_MemberData", (object)(new int[] { 4, 8 }))]
         // FailingMergeData has enumerables that throw errors when attempting to perform the nth enumeration.
         // This test checks whether the query runs in a pipelined or buffered fashion.
         public static void Merge_Ordered_Pipelining(Labeled<ParallelQuery<int>> labeled, int count, ParallelMergeOptions options)
         {
-            Assert.Equal(0, labeled.Item.WithMergeOptions(options).Select(x => x).First());
+            int partitions = Environment.ProcessorCount == 1 ? 1 : count - 1;
```

Comment:
Why the check for ProcessorCount?  What's special about ProcessorCount == 1 in this case?